newIDE/app/src/ObjectEditor/Editors/SpriteEditor/AnimationPreview/index.js

```diff
@@ -124,21 +126,37 @@ export default class AnimationPreview extends Component<Props, State> {
           project={project}
         />
         <Line noMargin alignItems="center">
+          <span style={styles.timeIcon}>FPS:</span>
+          <TextField
+            value={fps}
+            onChange={(e, text) => {
+              this.setState({fps: parseFloat(text)});
+              onChangeTimeBetweenFrames(parseFloat((1/text).toFixed(4)));
```

Comment:
Beware, doing `1/text` would result in NaN if text is not a parsable number, and parseFloat will return NaN too.

Prefer casting first to the proper type, and always always always do operation on numbers. Never rely on JavaScript doing conversion for you. It will backfire.
Also always consider your user inputs as "dirty" and do safety checks/sanitization.

```
               const fps = parseFloat(text);
               if (fps > 0) {
                 this.setState({ fps });
                 onChangeTimeBetweenFrames(parseFloat((1/fps).toFixed(4)));
               }
```